proto/cacheclient.proto

```diff
@@ -402,16 +402,125 @@ message _ListRemoveRequest {
 
 message _ListRemoveResponse {}
 
+/*
+  # ListFetch and ListRetain documentation and examples
+
+  # start and end are zero-based indexes, both are optional
+
+  # start is inclusive
+  # If start is not set, then the default is Unbounded, which is treated as 0
+  # Unbounded == 0,
+  # this field has a explicit presence for future compatibility to support reverse order and controlling our stride
+
+  # end is exclusive
+  # If end is not set, then the default is Unbounded which becomes (list.len)
+
+  # start and end can be negative numbers, negative indicates offsets starting at the end of the list.
+  # For example, -1 is the last element of the list, -2 the penultimate, etc
+  # list:                     [  a,  b,  c,  d,  e,  f,  g,  h,  i,  j]
+  # positive index positions:    0,  1,  2,  3,  4,  5,  6,  7,  8,  9
+  # negative index positions:  -10, -9, -8, -7, -6, -5, -4, -3, -2, -1
+
+  list_ = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+  # Using the slice notation:
+  # start:end
+
+  # start=0, end=Unbounded
+  list_[0:] == [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+  # start=5, end=Unbounded
+  list_[5:] == [5, 6, 7, 8, 9]
+
+  # start=5, end=0
+  list_[5:0] == []
+
+  # start=-5, end=Unbounded
+  list_[-5:] == [5, 6, 7, 8, 9]
+
+  # start=-5, end=0
+  list_[-5:0] == []
+
+  # start=0, end=5
+  list_[0:5] == [0, 1, 2, 3]
+
+  # start=Unbounded, end=5
+  list_[0:5] == [0, 1, 2, 3]
+
+  # start=0, end=-5
+  list_[0:-5] == [0, 1, 2, 3, 4]
+
+  # start=Unbounded, end=-5
+  list_[0:-5] == [0, 1, 2, 3, 4]
+
+  # start=2, end=6
+  list_[2:6] == [2, 3, 4, 5]
+
+  # start=2, end=-6
+  list_[2:-6] == [2, 3]
+
+  # start=-7, end=-2
+  list_[-7:-2] == [3, 4, 5, 6, 7]
+
+  // if using ListRetain, this will remove all elements in the list
+  # start=-2, end=-6
+  list_[2:-6] == []
+
+  # empty list
+  list_[0:0] == []
+
+  # List of length one special cases
+  list_size_one_ = ['a']
+
+  # start=-1, end=1
+  list_size_one_[-1:1] == ['a']
+
+  # start=1, end=-1
+  list_size_one_[1:-1] == []
+
+  # start=0, end=0
+  list_size_one_[0:0] == []
+
+  # start=0, end=-1
+  list_size_one_[-1:0] == []
+ */
+message _Unbounded { }
+
```

Comment:
Could you merge from main and clean up https://github.com/momentohq/client-protos/blob/main/proto/cacheclient.proto#L469 ? Also you'll have to build again with what's in main to get `client_protos.dsc` correctly updated.